riscv-pages/src/state.rs

```diff
@@ -111,3 +111,9 @@ impl InternalData for InternalDirty {}
 impl Cleanable for InternalDirty {
     type Cleaned = InternalClean;
 }
+
+/// A mapped page that is eligible for conversion to a Shared page
+#[derive(Debug)]
+pub enum Shareable {}
```

Comment:
I wonder if there should be two states here: `Mapped` pages that are currently mapped in a page table but haven't yet been marked as shared, and `MappableShared` for `Mapped` pages that have been marked as shared in `PageTracker`.

Reply:
Possibly, but likely doesn't belong in this MR?

Reply:
I think this has been resolved, but as mentioned below, it has been very difficult for me to infer the intent of the various traits and the interactions from the code.